common/core/web/keyboard-processor/tests/cases/transcriptions.js

```diff
@@ -55,11 +66,58 @@ describe("Transcriptions and Transforms", function() {
       assert.equal(transcription.transform.deleteRight, 0);
     });
 
+    it("handles operations with moderately long text", function() {
+      var Mock = com.keyman.text.Mock;
+
+      var target = new Mock("The quick brown cat jumped onto the lazy dog.", 19);
+      var original = Mock.from(target);
+      target.setDeadkeyCaret(30); // 19 + 11:  moves it to after "onto".
+      target.deleteCharsBeforeCaret(14); // delete:  "cat jumped onto"
+      target.insertTextBeforeCaret("fox jumped over");
+
+      /* It's not exactly black box, but presently we don't NEED the keyEvent object for the method to work.
+      * Other modules coming later will need it, though.
+      */
+      var transcription = target.buildTranscriptionFrom(original, null);
+
+      assert.equal(transcription.transform.insert, "fox jumped over", "Reported inserted text when only deletions exist");
+      assert.equal(transcription.transform.deleteLeft, 3, "Incorrect count for left-of-caret deletions");
+      assert.equal(transcription.transform.deleteRight, 11, "Incorrect count for right-of-caret deletions");
+    });
+
+    it("handles operations with long text", function() {
+      var Mock = com.keyman.text.Mock;
+
+      // Eh... had to pick SOMETHING.
+      let text = `Did you ever hear the Tragedy of Darth Plagueis the wise? I thought not.
+It's not a story the Jedi would tell you. It's a Sith legend. Darth Plagueis was a
+Dark Lord of the Sith, so powerful and so wise he could use the Force to influence
+the midichlorians to create life... He had such a knowledge of the dark side that
+he could even keep the ones he cared about from dying. The dark side of the Force
+is a pathway to many abilities some consider to be unnatural. He became so powerful...
+the only thing he was afraid of was losing his power, which eventually, of course,
+he did. Unfortunately, he taught his apprentice everything he knew, then his
+apprentice killed him in his sleep. It's ironic he could save others from death,
+but not himself.`;  // Sheev Palpatine, in the Star Wars prequels.
```

Comment:
Had to pick _something_ to use for a "long text" test case.

When I copied this 512 times over, the unit test took near a second to run.  Somehow, for both the original and the new versions.  (I definitely expected longer for the 'original'.)  Not sure if there was some odd caching effect happening or what.

Using a _single_ copy of this as the initial value of the "robust testing" page's text area works quite well for interactively testing the change.  Simply insert within the element here, between the tags:

https://github.com/keymanapp/keyman/blob/6abc0855e0c9f5103195c1fc6c447052b7d64e35/web/testing/prediction-mtnt/index.html#L90